core/tests/unit/ldml/test_transforms.cpp

```diff
@@ -1033,6 +1033,44 @@ test_normalize() {
   return EXIT_SUCCESS;
 }
 
+#include <unicode/uchar.h>
+
+#define NFD_FILE "common/web/types/src/ldml-keyboard/nfd-table.ts"
+int
+write_nfd_table() {
+  std::cout << " writing: " << NFD_FILE << std::endl;
+  FILE *f = fopen(NFD_FILE, "w");
+  assert(f != nullptr);
+  // we're going to need an NFD normalizer
+  UErrorCode status           = U_ZERO_ERROR;
+  const icu::Normalizer2 *nfd = icu::Normalizer2::getNFDInstance(status);
+  assert(U_SUCCESS(status));
+  fprintf(f, "// NFD hasBoundaryBefore\n");
+  fprintf(f, "// Generated by test_transforms.cpp\n");
+  fprintf(f, "// regenerate with: core/build/SOMETHING/SOMETHING/tests/unit/ldml/test_transforms --write-nfd\n");
+  fprintf(f, "export const ICU_VERSION='%s';\n", U_ICU_VERSION);
+  fprintf(f, "export const UNICODE_VERSION='%s';\n", U_UNICODE_VERSION);
+
+  fprintf(f, "export const nfdNoBoundaryBefore = [\n");
+  for (km_core_usv ch = 0; ch < 0x10FFFF; ch++) {
+    bool bb = nfd->hasBoundaryBefore(ch);
+    icu::UnicodeString s;
+    s.append((UChar32)ch);
+    bool lccc = nfd->isNormalized(s, status) && u_getCombiningClass(ch) == 0;
+    assert(U_SUCCESS(status));
+    if (bb != lccc) {
+      printf("0x%04x - bb=%s but lccc=%s\n", (unsigned int)ch, bb ? "y" : "n", lccc ? "y" : "n");
+    }
+    assert(bb == lccc);
```

Comment:
If this is an error condition then it really should fail in all cases, not just print a warning. asserts only fail in debug builds.

Reply:
this is a test, so don't asserts always fail in tests?

Reply:
This is not really a test, it's a build process, and while it is currently hosted within a unit test executable, that will hopefully change, at which point we want to be aborting here even on a release build.